src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -1335,10 +1338,24 @@ def get_secret_store(cli_ctx, name):
 
 def should_encrypt_token_cache(cli_ctx):
     # Only enable encryption for Windows (for now).
+    if cli_ctx.is_spython:
+        return False
+
     fallback = sys.platform.startswith('win32')
 
     # EXPERIMENTAL: Use core.encrypt_token_cache=False to turn off token cache encryption.
     # encrypt_token_cache affects both MSAL token cache and service principal entries.
     encrypt = cli_ctx.config.getboolean('core', 'encrypt_token_cache', fallback=fallback)
 
     return encrypt
+
+
+def is_spython():
+    return 'IS_SPYTHON' in os.environ
+
+
+def get_http_transport():
+    if is_spython():
+        from azure.cli.core.windows_http_transport import WindowsHttpTransport
+        return WindowsHttpTransport()
+    return None
```

Comment:
I think we can move this evaluation logic to `client_factory.py`.

Reply:
I put the logic here so other client may also take advantage of it.  For instance: https://github.com/Azure/azure-cli/blob/9d6ef592a3052d562f795455416b3bf2a81575c3/src/azure-cli/azure/cli/command_modules/storage/_client_factory.py#L258